### DISCUSSION TITLE
osu!catch shows cursor on the left/right edges of the screen

### DISCUSSION THREAD

**Discussion by Mogiiii** [ppy/osu#21645] [Q&A] [answered]
### Type

Cosmetic

### Bug description

osu!catch now hides the cursor on the playfield ( #21625 ) but since the playfield only covers the middle ~half of the screen the cursor still shows up on the edges. Since `CatchPlayField.RecievePositionalInputAt` already doesn't care Y axis, we can modify it to not care about the X axis either to "solve" the issue. I.E. `RecievePositionalInputAt` always returns `true`. This type of function returning a constant feels like a hack more than a proper solution to me, so I am opening this issue to solicit feedback.

```
public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) =>
     // only check the X position; handle all vertical space.
     base.ReceivePositionalInputAt(new Vector2(screenSpacePos.X, ScreenSpaceDrawQuad.Centre.Y));
```

This should have been fixed as part of #21625 but I thought this was an issue with the touchscreen controls rather than the playfield. I have since discovered otherwise.

### Screenshots or videos

https://user-images.githubusercontent.com/7416782/207783873-1aedff1b-e49b-48f9-a79c-d100f2ed68ff.mp4

### Version

60c8ef3fe5fe9adcb204814d975b7ed6c832fc8c

### Logs

N/A

**Comment by peppy** [comment, marked as answer]
This would make on-screen controls inaccessible. Could you better describe your use case for needing the mouse to be hidden on the edges when it is already hidden every time there's a keyboard input?

**Reply by Mogiiii** [reply]
In my local build, touchscreen controls still responded to mouse. I don't have a touch screen device set up for debug to test with proper touches.

As for the use case: this relates back to #21088, anyone using mouse to dash will still get a cursor and click sound on the screen edges. I would expect their mouse to not be perfectly still during gameplay, and could easily drift out of the current playfield and again become a problem.
